### ISSUE TITLE
TENSOR FLOW CNN MNIST EXAMPLE: HOW BATCH SIZE WORKS IN THE MODEL

### ISSUE THREAD

**Issue by Daromog** [tensorflow/tensorflow#20135]
In the CNN MNIST example of tensorflow I do not understand how batch size works, when they call the model they specify the size of the bach in 100:

`train_input_fn = tf.estimator.inputs.numpy_input_fn(
x={"x": train_data},
y=train_labels,
batch_size=100,
num_epochs=None,shuffle=True)
mnist_classifier.train(input_fn=train_input_fn,steps=20000,hooks=[logging_hook])`

But when the model is called

`def cnn_model_fn(features, labels, mode):
  input_layer = tf.reshape(features["x"], [-1, 28, 28, 1])`

They put -1 in batch size , I read in tensorflow tutorials and -1 they used it when they told the computer to infer that dimension What I do not understand is that before we put 100 and now because -1 do not understand how is entering the batch size to the model could you help me explaining? Thank you.

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by drpngx** [comment]
This question is better asked on [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there.

If you think we've misinterpreted a bug, please comment again with a clear explanation, as well as all of the information requested in the [issue template](https://github.com/tensorflow/tensorflow/issues/new). Thanks!
